File: exercises/D702/solution.c
#include <stdio.h>
int st[1000];
char s[10];
int main() {
    int n, x, i;
    int size = 0;
    scanf("%d", &n);
    for (i = 0; i < n; i++) {
        scanf("%s", s);
        if (strcmp(s, "PUSH") == 0) {
            scanf("%d", &x);
            st[size++] = x;
        } else if (strcmp(s, "TOP") == 0) {
            if (size == 0) {
                printf("Empty\n");
            } else {
                printf("%d\n", st[size - 1]);
            }
        } else if (strcmp(s, "POP") == 0) {
            if (size == 0) {
                printf("Cannot pop\n");
            } else {
                size--;
            }
        } else {
            printf("%d\n", size);
        }
    }
    return 0;
}

File: exercises/D702/solution.java
import java.util.*;

class solution {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int[] st = new int[1000];
        int size = 0;
        int n = in.nextInt();
        for (int i = 0; i < n; i++) {
            String s = in.next();
            if (s.equals("PUSH")) {
                int x = in.nextInt();
                st[size++] = x;
            } else if (s.equals("TOP")) {
                if (size == 0) {
                    System.out.println("Empty");
                } else {
                    System.out.println(st[size - 1]);
                }
            } else if (s.equals("POP")) {
                if (size == 0) {
                    System.out.println("Cannot pop");
                } else {
                    size--;
                }
            } else {
                System.out.println(size);
            }
        }
    }
}
